frontend/src/components/Dropdown/SingleSelect/SingleSelectProvider.tsx

```diff
@@ -0,0 +1,129 @@
+import { useCallback, useMemo, useState } from 'react'
+import { FormControlOptions, useMultiStyleConfig } from '@chakra-ui/react'
+import { useCombobox } from 'downshift'
+
+import { useItems } from '../hooks/useItems'
+import { SelectContext, SharedSelectContextReturnProps } from '../SelectContext'
+import { ComboboxItem } from '../types'
+import { defaultFilter } from '../utils/defaultFilter'
+import { itemToLabelString, itemToValue } from '../utils/itemUtils'
+
+export interface SingleSelectProviderProps<
+  Item extends ComboboxItem = ComboboxItem,
+> extends SharedSelectContextReturnProps<Item>,
+    FormControlOptions {
+  /** Controlled input value */
+  value: string
+  /** Controlled input onChange handler */
+  onChange: (value: string) => void
+  /** Function based on which items in dropdown are filtered. Default filter filters by fuzzy match. */
+  filter?(items: Item[], value: string): Item[]
+  /** Initial dropdown opened state. Defaults to `false`. */
+  defaultIsOpen?: boolean
+  children: React.ReactNode
+}
+export const SingleSelectProvider = ({
```

Comment:
Actually, there shouldn't be a need to expose the generic types, since the API of the dropdown component does not care. 

Generic types are there for ease of refactoring if we ever want to in the future